basketball_reference_web_scraper/client.py

```diff
@@ -30,6 +30,37 @@ def player_box_scores(day, month, year, output_type=None, output_file_path=None,
     )
 
 
+def player_season_box_score(player_name, season_end_year, url_override='01', output_type=None, output_file_path=None, output_write_option=None,
+                             json_options=None):
+
+    # TODO (url_override hax):
+    # the bballref url link for a player's season box score contains:
+    #    <first 5 letters of last name><first 2 letters of first name><duplication id>
+    # the duplication id defaults to 01, but if there are two or more players that
+    # have similar names (e.g. Jabari Brown and Jaylen Brown are both brownja),
+    # then the duplication id is what sets them apart (i.e. brownja01 & brownja02).
+    # allow the user to manually pass this in for now until we can find a better
+    # method of resolving the correct id.
```

Comment:
A possible solution to this is to continuously make requests with incrementing numbers until we find a response that has a matching player name (this would need to get parsed as well). Only thing is this would fail for cases where there were two players with the exact same name, especially if both are still active in the NBA. Then there's the issue of repeating this work every time a request is made, as opposed to caching the result somewhere so we aren't making extra unnecessary requests. Curious to hear your thoughts on this.

Reply:
@ntsirakis it looks like there's a search endpoint that you could also use to try and identify the correct player

For example, take the case of Jabari Brown

`https://www.basketball-reference.com/search/search.fcgi?hint=jabari+brown&search=jabari+brown&pid=&idx=`

![image](https://user-images.githubusercontent.com/8136030/72809717-2e38f880-3c2a-11ea-9442-bcd343a15b55.png)

You could iterate through all returned search results and select the first exact match for a given name (or return _every_ exact match to deal with the case where there are multiple players with the same name).

Just inspecting the search results web page, it looks like the player URL is exposed for each player result.

Reply:
I think there's no way around it; that's probably the most reliable method. The only issue I can see with that is that we're making a double http request per boxscore request, which may or may not be inefficient when dealing with a lot of data. That would normally be a problem for my particular project (which iterates over many players for each calculation), but I cache all my results once fetched, so it's not an issue. Do you want me to add that before merge or should I leave the TODO for now?

Reply:
@ntsirakis I definitely understand the concerns around network cost but I think I'd prefer to provide as reliable an API as possible and then start trying to find ways to optimize.

I can take care of making this change if you could try and resolve as many of the merge conflicts as possible.

Then, I'll probably merge to a feature branch and do any additional clean-up before merging into the `v4` branch.